garage/tf/policies/gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,158 @@
+"""GaussianMLPPolicy with GaussianMLPModel."""
+import numpy as np
+import tensorflow as tf
+
+from garage.misc import logger
+from garage.tf.distributions import DiagonalGaussian
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+from garage.tf.spaces import Box
+
+
+class GaussianMLPPolicyWithModel:
+    """
+    GaussianMLPPolicy with GaussianMLPModel.
+
+    :param env_spec:
+    :param hidden_sizes: list of sizes for the fully-connected hidden
+    layers
+    :param learn_std: Is std trainable
+    :param init_std: Initial std
+    :param adaptive_std:
+    :param std_share_network:
+    :param std_hidden_sizes: list of sizes for the fully-connected layers
+     for std
+    :param min_std: whether to make sure that the std is at least some
+     threshold value, to avoid numerical issues
+    :param std_hidden_nonlinearity:
+    :param hidden_nonlinearity: nonlinearity used for each hidden layer
+    :param output_nonlinearity: nonlinearity for the output layer
+    :param mean_network: custom network for the output mean
+    :param std_network: custom network for the output log std
+    :param std_parametrization: how the std should be parametrized. There
+     are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+         exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+    :return:
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 name='GaussianMLPPolicy',
+                 dist=DiagonalGaussian,
```

Comment:
by definition, doesn't a GaussianMLPPolicy use a DiagonalGaussian? I'm confused about why this is a parameter at all.